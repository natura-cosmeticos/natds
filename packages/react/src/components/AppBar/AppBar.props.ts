export type AppBarProps = {
  children?: React.ReactNode
  /**
   * Override or extend the styles applied to the component
   */
  className?: string
  /**
   * Override or extend the styles applied to the component
   */
  color?: 'default' | 'primary' | 'none' | 'inverse'
  /**
   * If `false`, a box-shadow effect are disabled
   * @default true
   */
  elevation?: boolean
  /**
   * Optional ID for testing
   */
  testID?: string
}
