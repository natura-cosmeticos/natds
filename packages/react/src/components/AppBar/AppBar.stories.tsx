/* eslint-disable @typescript-eslint/ban-ts-comment */
import React from 'react'
import { Story, Meta } from '@storybook/react'
import { IconButton } from '../IconButton'
import { Icon } from '../Icon'
import { AppBar, AppBarProps } from '.'
import { Logo } from '../Logo'

const componentStatus = `
---

**NOTE FOR UX**: This component is available in the following variants:

  - ✅ \`base\`
  - ✅ \`standard\`

With the following attribute status:

  - **Color**
      - ✅ \`default\`
      - ✅ \`primary\`
      - ✅ \`inverse\`
      - ✅ \`none\`
  - ✅ **Elevation**
  - ✅ **Content Position** (available but with alternative composition)
  - ✅ **Content Type** (available but with alternative composition)
  - ✅ **Content Proeminent** (available but with alternative composition)
  - **Behaviour**
      - ✅  \`fixed\`
      - ✅  \`scroll\`
  - **Action**
      - ✅ \`left\` (available but with alternative composition)
      - ✅ \`right\` (available but with alternative composition)

---
`

export default {
  title: 'Components/AppBar',
  component: AppBar,
  parameters: {
    componentSubtitle: 'The App Bar Top displays information and actions relating to the current screen.',
    docs: { description: { component: componentStatus } }
  }
} as Meta

export const Playground: Story<AppBarProps> = (args) => <AppBar {...args} />
Playground.args = {
  children: 'App Bar - Top',
  // @ts-ignore
  style: { position: 'static' }
}

export const Color: Story<AppBarProps> = (args) => (
  <div style={{ display: 'flex', gap: 8, flexDirection: 'column' }}>
    <AppBar {...args} color="default">Default</AppBar>
    <AppBar {...args} color="primary">Primary</AppBar>
    <AppBar {...args} color="inverse">Inverse</AppBar>
    <AppBar {...args} color="none">None</AppBar>
  </div>
)

Color.args = {
  ...Playground.args
}

export const Elevation: Story<AppBarProps> = (args) => (
  <div style={{ display: 'flex', gap: 8, flexDirection: 'column' }}>
    <AppBar {...args}>Elevation</AppBar>
    <AppBar {...args} elevation={false}>No Elevation</AppBar>
  </div>
)

Elevation.args = {
  ...Playground.args
}

export const WithLogo: Story<AppBarProps> = (args) => (
  <AppBar {...args}>
    <div style={{ display: 'flex', alignItems: 'center', width: '100%' }}>
      <IconButton IconComponent={<Icon name="outlined-navigation-menu" color="highlight" />} ariaLabel="menu" onClick={() => ''} />
      <div style={{ flexGrow: 1, justifyContent: 'center', display: 'flex' }}>
        <Logo size="hugeX" />
      </div>
      <>
        <IconButton IconComponent={<Icon name="outlined-action-search" color="highlight" />} ariaLabel="search" onClick={() => ''} />
        <IconButton IconComponent={<Icon name="outlined-finance-bag" color="highlight" />} ariaLabel="profile" onClick={() => ''} />
      </>
    </div>
  </AppBar>
)
WithLogo.args = {
  ...Playground.args
}
