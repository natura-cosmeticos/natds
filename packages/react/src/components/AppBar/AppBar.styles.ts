import { createUseStyles } from 'react-jss'
import type { Theme } from '@naturacosmeticos/natds-themes'
import type { AppBarProps } from './AppBar.props'

const styles = createUseStyles((theme: Theme) => ({
  appBar: {
    alignItems: 'center',
    display: 'flex',
    flex: 'auto',
    flexDirection: 'row',
    justifyContent: 'space-between',
    minHeight: theme.size.mediumX,
    paddingLeft: theme.spacing.tiny,
    paddingRight: theme.spacing.tiny,
    width: 'auto',
    background: ({ color }: AppBarProps) => {
      switch (color) {
        case 'primary':
          return theme.color.primary
        case 'inverse':
          return theme.color.highEmphasis
        case 'none':
          return 'none'
        default:
          return theme.color.surface
      }
    },
    boxShadow: ({ elevation }: AppBarProps) => (elevation ? theme.elevation.tiny : 'none'),
    fontFamily: [theme.appBarTop.title.primary.fontFamily, theme.appBarTop.title.fallback.fontFamily],
    fontSize: theme.appBarTop.title.fontSize,
    fontWeight: theme.appBarTop.title.primary.fontWeight,
    letterSpacing: theme.appBarTop.title.letterSpacing,
    lineHeight: theme.appBarTop.title.lineHeight
  }
}))

export default styles
