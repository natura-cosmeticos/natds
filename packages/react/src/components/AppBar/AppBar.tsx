import React from 'react'
import styles from './AppBar.styles'
import type { AppBarProps } from './AppBar.props'

const AppBar = React.forwardRef<HTMLDivElement, AppBarProps>(({
  className = '',
  color = 'default',
  elevation = true,
  testID,
  children
},
ref) => {
  const { appBar } = styles({ elevation, color })

  return (
    <div className={`${className} ${appBar}`} data-testid={testID} ref={ref}>
      {children}
    </div>
  )
})

export default AppBar
