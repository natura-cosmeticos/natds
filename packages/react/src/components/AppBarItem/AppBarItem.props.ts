export type AppBarItemProps = {
  children?: React.ReactNode
  /**
   * Override or extend the styles applied to the component
   */
  className?: string
  /**
   * Optional ID for testing
   */
  testID?: string
}
