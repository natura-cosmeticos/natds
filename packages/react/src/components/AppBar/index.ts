import AppBar from './AppBar'

export { AppBar }
export { AppBarProps } from './AppBar.props'
export default AppBar
