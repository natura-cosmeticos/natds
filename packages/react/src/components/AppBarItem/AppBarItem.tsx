import React from 'react'
import { AppBarItemProps } from './AppBarItem.props'
import styles from './AppBarItem.styles'

const AppBarItem = React.forwardRef<HTMLDivElement, AppBarItemProps>((props, ref) => {
  const { className = '', testID, ...rest } = props
  const { appBarItem } = styles()

  return <div {...rest} className={`${className} ${appBarItem}`} data-testid={testID} ref={ref} />
})

export default AppBarItem
