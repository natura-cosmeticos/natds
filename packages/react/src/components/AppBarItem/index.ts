import AppBarItem from './AppBarItem'

export { AppBarItem }
export { AppBarItemProps } from './AppBarItem.props'
export default AppBarItem
