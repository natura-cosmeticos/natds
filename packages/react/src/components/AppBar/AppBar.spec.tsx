import React from 'react'
import AppBar from '.'
import renderWithTheme from '../../helpers/renderWithTheme'
import AppBarItem from '../AppBarItem'

describe('AppBar component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<AppBar>app bar</AppBar>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly without elevation', () => {
    const { styles, component } = renderWithTheme(<AppBar elevation={false}>app bar</AppBar>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with primary color variant', () => {
    const { styles, component } = renderWithTheme(<AppBar color="primary">app bar</AppBar>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with inverse color variant', () => {
    const { styles, component } = renderWithTheme(<AppBar color="inverse">app bar</AppBar>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with none color variant', () => {
    const { styles, component } = renderWithTheme(<AppBar color="none">app bar</AppBar>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
