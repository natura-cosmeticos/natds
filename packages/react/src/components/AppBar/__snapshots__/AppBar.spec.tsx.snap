// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component should render correctly with default props 1`] = `
Array [
  ".container-0-2-1 {}
.appBar-0-2-2 {
  flex: auto;
  width: auto;
  display: flex;
  padding: 4px 8px;
  font-size: 20px;
  min-height: 56px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  line-height: 1.25;
  flex-direction: row;
  letter-spacing: 0.24px;
  justify-content: space-between;
}
.container-d0-0-2-3 {
  top: 0;
  left: 0;
  width: 100%;
  position: fixed;
}
.appBar-d1-0-2-4 {
  background-color: #ffffff;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  color: #333333;
}",
  <div>
    <div
      class="container-0-2-1 container-d0-0-2-3"
    >
      <div
        class=" appBar-0-2-2 appBar-d1-0-2-4"
      >
        app bar
      </div>
    </div>
  </div>,
]
`;

exports[`AppBar component should render correctly with inverse color variant 1`] = `
Array [
  ".container-0-2-1 {}
.appBar-0-2-2 {
  flex: auto;
  width: auto;
  display: flex;
  padding: 4px 8px;
  font-size: 20px;
  min-height: 56px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  line-height: 1.25;
  flex-direction: row;
  letter-spacing: 0.24px;
  justify-content: space-between;
}
.container-d0-0-2-3 {
  top: 0;
  left: 0;
  width: 100%;
  position: fixed;
}
.appBar-d1-0-2-4 {
  background-color: #333333;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  color: #ffffff;
}",
  <div>
    <div
      class="container-0-2-1 container-d0-0-2-3"
    >
      <div
        class=" appBar-0-2-2 appBar-d1-0-2-4"
      >
        app bar
      </div>
    </div>
  </div>,
]
`;

exports[`AppBar component should render correctly with none color variant 1`] = `
Array [
  ".container-0-2-1 {}
.appBar-0-2-2 {
  flex: auto;
  width: auto;
  display: flex;
  padding: 4px 8px;
  font-size: 20px;
  min-height: 56px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  line-height: 1.25;
  flex-direction: row;
  letter-spacing: 0.24px;
  justify-content: space-between;
}
.container-d0-0-2-3 {
  top: 0;
  left: 0;
  width: 100%;
  position: fixed;
}
.appBar-d1-0-2-4 {
  background-color: transparent;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  color: #000000;
}",
  <div>
    <div
      class="container-0-2-1 container-d0-0-2-3"
    >
      <div
        class=" appBar-0-2-2 appBar-d1-0-2-4"
      >
        app bar
      </div>
    </div>
  </div>,
]
`;

exports[`AppBar component should render correctly with primary color variant 1`] = `
Array [
  ".container-0-2-1 {}
.appBar-0-2-2 {
  flex: auto;
  width: auto;
  display: flex;
  padding: 4px 8px;
  font-size: 20px;
  min-height: 56px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  line-height: 1.25;
  flex-direction: row;
  letter-spacing: 0.24px;
  justify-content: space-between;
}
.container-d0-0-2-3 {
  top: 0;
  left: 0;
  width: 100%;
  position: fixed;
}
.appBar-d1-0-2-4 {
  background-color: #f4ab34;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14);
  color: #333333;
}",
  <div>
    <div
      class="container-0-2-1 container-d0-0-2-3"
    >
      <div
        class=" appBar-0-2-2 appBar-d1-0-2-4"
      >
        app bar
      </div>
    </div>
  </div>,
]
`;

exports[`AppBar component should render correctly without elevation 1`] = `
Array [
  ".container-0-2-1 {}
.appBar-0-2-2 {
  flex: auto;
  width: auto;
  display: flex;
  padding: 4px 8px;
  font-size: 20px;
  min-height: 56px;
  align-items: center;
  font-family: Roboto, Arial;
  font-weight: 500;
  line-height: 1.25;
  flex-direction: row;
  letter-spacing: 0.24px;
  justify-content: space-between;
}
.container-d0-0-2-3 {
  top: 0;
  left: 0;
  width: 100%;
  position: fixed;
}
.appBar-d1-0-2-4 {
  background-color: #ffffff;
  box-shadow: none;
  color: #333333;
}",
  <div>
    <div
      class="container-0-2-1 container-d0-0-2-3"
    >
      <div
        class=" appBar-0-2-2 appBar-d1-0-2-4"
      >
        app bar
      </div>
    </div>
  </div>,
]
`;
